Show lack of backend connection and retry login every 5 seconds

skaramicke opened the pull request.

Imamyable commented on app/navigation/RootStackNavigation.tsx:
```diff
@@ -1,30 +1,46 @@
+          <Stack.Screen
+            name="Start"
+            options={{
+              headerShown: !backendConnection,
```
Instead of replacing the loginpage we should show this as an alert instead

skaramicke replied:
This part is only the banner that appears when you're logged in and a request fails due to backend being unreachable.

Imamyable commented on app/App.tsx:
```diff
@@ -79,7 +101,15 @@ function App(): JSX.Element {
+        <Center w="100%" h="100%">
+          <Text>Det går inte att nå servern just nu.</Text>
+          <Text>
+            Försöker igen om {timeLeft} sekund{timeLeft !== 1 ? 'er' : ''}
```
place this info in the login page in an alert, and disable the button instead

skaramicke replied:
I agree when the user is initiating the login. It makes less sense when the app is trying to log in using credentials that are known to work. The user would be logged in automatically if the backend server was reachable, so I feel something like the current UX flow would be more appropriate in that scenario. However, I don't have a good idea of how to do that now, so I'm switching to the alert as suggested, because that's a much better single solution than the current one.

I've added issue #176 with a suggestion for that case.

skaramicke replied:
Wait, disabling the login form button is still a variant of my first suggestion. I'll need to either move the login api call to the login form or move the login form to the App component to get this working. I think I prefer the latter. I'll try that and see how it behaves.

skaramicke commented on app/navigation/RootStackNavigation.tsx:
```diff
@@ -1,30 +1,84 @@
+                : NoWifiIcon,
+            }}>
+            {() => {
+              if (isTryingToLogin) {
```
Perhaps this should be changed to remove LoadingScreen and just switch between LoggedInTabs and SignInForm based on user.

skaramicke commented:
Changes since last review:
- Moved automatic login stuff from App to SigninForm.
- App no longer process user state but shows RootStackNavigation directly, like in develop.
- Moved backend connection recheck from App to RootStackNavigation so it rechecks against the health endpoint every second, and changes the backend connection state to true if it succeeds. It indicates the process by showing a spinner while loading. Feels intuitive and informative to me.
- Locked the signin button while backend connection is false.
- Automatically logs in if you have stored credentials, when backend connection returns while on the login form.

Possible things to keep working on:
- If you check the save credentials box, log in, discover backend is offline and then realize you put the wrong password, you can't currently stop the form from posting when backend comes back. It'll try again and fail, and then you can change the details and try again. Perhaps a cancel button, or checking if the save credentials box is still checked, and if so not try again automatically.
